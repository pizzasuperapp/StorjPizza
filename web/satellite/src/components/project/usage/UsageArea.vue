// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="usage-area">
        <p class="usage-area__title">{{ title }}</p>
        <VLoader v-if="isDataFetching" />
        <template v-else>
            <pre class="usage-area__remaining">{{ remainingFormatted }} Remaining</pre>
            <VBar
                :current="used"
                :max="limit"
                color="#0068DC"
            />
            <div class="usage-area__limits-area">
                <pre class="usage-area__limits-area__title">{{ title }} Used</pre>
                <pre class="usage-area__limits-area__limits">{{ usedFormatted }} / {{ limitFormatted }}</pre>
            </div>
        </template>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import VBar from '@/components/common/VBar.vue';
import VLoader from '@/components/common/VLoader.vue';

import { Dimensions, Size } from '@/utils/bytesSize';

// @vue/component
@Component({
    components: {
        VBar,
        VLoader,
    },
})
export default class UsageArea extends Vue {
    @Prop({default: ''})
    public readonly title: string;
    @Prop({default: 0})
    public readonly used: number;
    @Prop({default: 0})
    public readonly limit: number;
    @Prop({default: true})
    public readonly isDataFetching: boolean;

    /**
     * Returns formatted remaining amount.
     */
    public get remainingFormatted(): string {
        const remaining = this.limit - this.used;
        if (remaining < 0) {
            return '0 Bytes';
        }

        return this.formattedValue(new Size(remaining, 2));
    }

    /**
     * Returns formatted used amount.
     */
    public get usedFormatted(): string {
        return this.formattedValue(new Size(this.used, 2));
    }

    /**
     * Returns formatted limit amount.
     */
    public get limitFormatted(): string {
        return this.formattedValue(new Size(this.limit, 2));
    }

    /**
     * Formats value to needed form and returns it.
     */
    private formattedValue(value: Size): string {
        switch (value.label) {
        case Dimensions.Bytes:
        case Dimensions.KB:
            return '0';
        default:
            return `${value.formattedBytes.replace(/\\.0+$/, '')}${value.label}`;
        }
    }
}
</script>

<style scoped lang="scss">
    h1,
    p,
    pre {
        margin: 0;
    }

    .usage-area {
        padding: 20px;
        border-radius: 6px;
        background-color: #fff;

        &__title {
            font-size: 16px;
            line-height: 16px;
            color: #1b2533;
            margin-bottom: 25px;
            font-family: 'font_bold', sans-serif;
        }

        &__remaining {
            font-size: 16px;
            line-height: 16px;
            color: #1b2533;
            margin-bottom: 15px;
            font-family: 'font_regular', sans-serif;
        }

        &__limits-area {
            width: 100%;
            display: flex;
            align-items: center;
            justify-content: space-between;
            margin-top: 15px;

            &__title {
                font-size: 14px;
                line-height: 14px;
                color: #384761;
                font-family: 'font_regular', sans-serif;
            }

            &__limits {
                font-size: 14px;
                line-height: 14px;
                color: #1b2533;
                font-family: 'font_regular', sans-serif;
            }
        }
    }
</style>
