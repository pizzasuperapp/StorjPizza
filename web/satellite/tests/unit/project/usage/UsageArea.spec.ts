// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import UsageArea from '@/components/project/usage/UsageArea.vue';

import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('UsageArea.vue', () => {
    it('renders correctly', (): void => {
        const wrapper = shallowMount(UsageArea, {
            localVue,
            propsData: {
                title: 'test Title',
                used: 500000000,
                limit: 1000000000,
                isDataFetching: false,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly if used > limit', (): void => {
        const wrapper = shallowMount(UsageArea, {
            localVue,
            propsData: {
                title: 'test Title',
                used: 1000000000,
                limit: 500000000,
                isDataFetching: false,
            },
        });

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.usage-area__remaining').text()).toMatch('0 Bytes Remaining');
    });
});
