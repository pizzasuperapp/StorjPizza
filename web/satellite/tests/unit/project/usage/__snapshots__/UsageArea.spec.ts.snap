// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsageArea.vue renders correctly 1`] = `
<div class="usage-area">
  <p class="usage-area__title">test Title</p> <pre class="usage-area__remaining">500.00MB Remaining</pre>
  <vbar-stub current="500000000" max="1000000000" color="#0068DC"></vbar-stub>
  <div class="usage-area__limits-area"><pre class="usage-area__limits-area__title">test Title Used</pre> <pre class="usage-area__limits-area__limits">500.00MB / 1.00GB</pre>
  </div>
</div>
`;

exports[`UsageArea.vue renders correctly if used > limit 1`] = `
<div class="usage-area">
  <p class="usage-area__title">test Title</p> <pre class="usage-area__remaining">0 Bytes Remaining</pre>
  <vbar-stub current="1000000000" max="500000000" color="#0068DC"></vbar-stub>
  <div class="usage-area__limits-area"><pre class="usage-area__limits-area__title">test Title Used</pre> <pre class="usage-area__limits-area__limits">1.00GB / 500.00MB</pre>
  </div>
</div>
`;
